db/sovereign-db/src/ledger_db/rpc.rs

```diff
@@ -78,19 +79,51 @@ impl LedgerRpcProvider for LedgerDB {
     ) -> Result<Vec<Option<Self::SlotResponse>>, anyhow::Error> {
         // TODO: https://github.com/Sovereign-Labs/sovereign/issues/191 Sort the input
         //      and use an iterator instead of querying for each slot individually
-        let mut out = Vec::with_capacity(slot_ids.len());
-        for slot_id in slot_ids {
-            let slot_num = self.resolve_slot_identifier(slot_id)?;
-            out.push(match slot_num {
-                Some(num) => {
-                    if let Some(stored_slot) = self.db.get::<SlotByNumber>(&num)? {
-                        Some(self.populate_slot_response(num.into(), stored_slot, query_mode)?)
-                    } else {
-                        None
+
+        // First resolve the slot identifiers and store the slot numbers
+        let mut slot_nums: Vec<Option<SlotNumber>> = slot_ids.iter()
+            .map(|slot_id| self.resolve_slot_identifier(slot_id))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        // Then sort the slot numbers
+        slot_nums.sort_unstable();
```

Comment:
Currently, this logic will return results in sorted order rather than the requested order. What if we keep track of the original indices and return the results in that order too. Something like this:

```rust
     let mut slot_nums: Vec<Option<SlotNumber>> = slot_ids.iter().enumerate().
            .map(|(idx, slot_id)| (idx, self.resolve_slot_identifier(slot_id)))
            .collect::<Result<Vec<_>, _>>()?;

     slot_nums.sort_unstable_by(|(_, slot_id_a), (_, slot_id_b)| slot_id_a.cmp(slot_id_b));

    // ...
    for (orig_idx, (start, end)) in slot_ranges {
            let stored_slots = self.db.get_range::<SlotByNumber>(start, end)?;
            for num in start..=end {
                if let Some(stored_slot) = stored_slots.get(&num) {
                    out[orig_idx] = Some(self.populate_slot_response(num.into(), stored_slot.clone(), query_mode)?);
                } else {
                    out[orig_idx]  = None;
                }
            }
        }

```